diff --git a/dts/arm/renesas/gen4/rcar_gen4_cr52.dtsi b/dts/arm/renesas/gen4/rcar_gen4_cr52.dtsi
index e8bc026ae1..04602bd645 100644
--- a/dts/arm/renesas/gen4/rcar_gen4_cr52.dtsi
+++ b/dts/arm/renesas/gen4/rcar_gen4_cr52.dtsi
@@ -107,7 +107,6 @@
 			clocks = <&cpg CPG_MOD 518>;
 			status = "disabled";
 			label = "i2c0";
-			status = "okay";
 		};
 
 		i2c1: i2c@e6508000 {
@@ -122,7 +121,6 @@
 			clocks = <&cpg CPG_MOD 519>;
 			status = "disabled";
 			label = "i2c1";
-			status = "okay";
 		};
 
 		i2c4: i2c@e66d8000 {
@@ -137,7 +135,6 @@
 			clocks = <&cpg CPG_MOD 522>;
 			status = "disabled";
 			label = "i2c4";
-			status = "okay";
 		};
 
 		i2c5: i2c@e66e0000 {
@@ -152,7 +149,6 @@
 			clocks = <&cpg CPG_MOD 523>;
 			status = "disabled";
 			label = "i2c5";
-			status = "okay";
 		};
 
 		eth0: ethernet@e6800000 {
@@ -2009,7 +2005,7 @@
 		    interrupt-parent = <&gic>;
 		    interrupts = <GIC_SPI 523 IRQ_TYPE_LEVEL 0>, <GIC_SPI 524 IRQ_TYPE_LEVEL 0>;
 		    label = "du_00";
-		    status = "okay";
+		    status = "disabled";
 		    clocks = <&cpg CPG_MOD 411>;
 		    resets = <&cpg 411>;
 		    power-domains = <&sysc POWER_DOMAIN_ALWAYS_ON>;
